### ISSUE TITLE
Installing javabridge - did not complete

### ISSUE THREAD

**Issue by makotojohira** [SchmollerLab/Cell_ACDC#382]
### Description:

I have installed acdc via Anaconda onto my MacBook Air with MacOS v 13.5.  The Welcome page opened and when I tried the Wizard button, the Installing javabridge window opened, a circling icon appeared, but the installation did not appear to complete (there was no dialog or popup indicating installation was complete).  I tried the Wizard button again, and the installing javabridge window popped open again.  I tried this several times before reporting the installation failure here.  Thank you.

**Comment by ElpadoCan** [comment]
Hi @makotojohira, I'm sorry to hear you are having these issues. Do you have a Apple Silicon Mac (M1 or M2) or a Intel based Mac?

Unfortunately, as far as I know, javabridge doesn't work on mac M1 or M2 and I don't think the developer will ever fix it.

We are working on an alternative but it's going to take some time. In the meantime, you will have to create the data structure manually or with a Fiji macro.

If you are familiar with the ImageJ macro language, we already have two macros on our GitHub page [here] (https://github.com/SchmollerLab/Cell_ACDC/tree/main/FijiMacros) that would create the data structure starting from the microscopy file.

The other option is that you convert the files to Tif and then I can add a small utility to rearrange the files into the required data structure.

Let me know what you prefer to do, thanks!

Cheers
Francesco

**Comment by makotojohira** [comment]
Hi Francesco,

Thanks for your quick reply.  I am using an Intel based Mac.  And thank you for the link to the Fiji macros.  My image is already a .tif file.  Should I use your macro on my .tif images?

Thanks again,

Makoto Ohira
Rhind Lab
https://www.umassmed.edu/rhindlab/
Department of Biochemistry and Molecular Biotechnology
University of Massachusetts Chan Medical School
LRB 940D
364 Plantation Street
Worcester, MA 01655

________________________________

**Comment by ElpadoCan** [comment]
Ok if it's tif you don't need javabridge nor the Fiji macros since they are both for opening raw microscopy files and convert them into tif.

However, we still need to create the data structure required by acdc.

How many dimensions do you have in your tif file? Are they 2D over time? Or z-stacks? Or z-stacks over time? How many tif files do you end up having in a typical experiment?

The data structure required is a single tif file per channel. The tif file can be a single 2D, 3D (z-stack or 2D over time), or 4D (3D z-stacks over time). Then you can have multiple positions per experiment. A typical scenario is to start from a single cell (each cell is a position) and image over time (the cell grows and divide generating a colony).

Anyway, as soon as I know how you have the tif files arranged we easily rearrange them into acdc structure.

If you think it can help, you can also share a sample of the dataset on Google drive and share it to my email padovaf@tcd.ie

Let me know thanks! 

Cheers
Francesco

**Comment by makotojohira** [comment]
Hi Francesco,


When I open acdc, I have two windows open.  One is Welcome window, the other is Cell-ACDC window.


In the Welcome window, when I click on Launch Wizard, it automatically opens the Installing javabridge window and when I click OK, it stalls with the rotating rainbow circle and never installs the program.


When I click on Test segmentation with my image/video, it opens Cell-ACDC - GUI, and if I click on the page + button, the Load image file button (after putting the .tif file in a folder called Image), it ultimately gives me a critical error window, and tells me to restart ACDC.


I do not have z-stacks or time-lapse image series - but a lab mate who will ultimately use acdc has those images.  I am just trying to install and test the acdc program.


With a single, simple 1-channel .tif image that I would like to upload and run a test segmentation on, just to demonstrate to myself that I have everything I need for my lab mates to use acdc, how can I get past these error messages?


Many thanks!


Makoto


Makoto Ohira
Rhind Lab
https://www.umassmed.edu/rhindlab/
Department of Biochemistry and Molecular Biotechnology
University of Massachusetts Chan Medical School
LRB 940D
364 Plantation Street
Worcester, MA 01655

________________________________

**Comment by ElpadoCan** [comment]
Ok great, then I need the error messages you get. You can either send me the log files or copy paste from the terminal.

One thing that I can say is that the tif file(s) should be in a folder called `Images` and not  `Image`.

For the `javabridge` installation instead in principle it should work on an Intel mac but I don't have one (I have only a M1 Mac) myself. I can send you detailed instructions if you want to give it a try.

However, while you are just testing out, if I was you, I would not try to install `javabridge` since it's needed only to open raw microscopy files and covert them to tif (and you already have tif files).

We are trying to migrate away from javabridge anyway because it's a pain to install and there seems to be better options now.

Cheers
Francesco

**Comment by makotojohira** [comment]
OK thanks again Francesco.  I have copied the following error messages from Terminal:

##############


WARNING: Cell-ACDC is in error state. Please, restart.

====================================

Closing GUI logger...

Traceback (most recent call last):

  File "/Users/makotoohira/miniconda3/envs/acdc/lib/python3.9/site-packages/urllib3/connectionpool.py", line 714, in urlopen

    httplib_response = self._make_request(

  File "/Users/makotoohira/miniconda3/envs/acdc/lib/python3.9/site-packages/urllib3/connectionpool.py", line 403, in _make_request

    self._validate_conn(conn)

  File "/Users/makotoohira/miniconda3/envs/acdc/lib/python3.9/site-packages/urllib3/connectionpool.py", line 1053, in _validate_conn

    conn.connect()

  File "/Users/makotoohira/miniconda3/envs/acdc/lib/python3.9/site-packages/urllib3/connection.py", line 419, in connect

    self.sock = ssl_wrap_socket(

  File "/Users/makotoohira/miniconda3/envs/acdc/lib/python3.9/site-packages/urllib3/util/ssl_.py", line 449, in ssl_wrap_socket

    ssl_sock = _ssl_wrap_socket_impl(

  File "/Users/makotoohira/miniconda3/envs/acdc/lib/python3.9/site-packages/urllib3/util/ssl_.py", line 493, in _ssl_wrap_socket_impl

    return ssl_context.wrap_socket(sock, server_hostname=server_hostname)

  File "/Users/makotoohira/miniconda3/envs/acdc/lib/python3.9/ssl.py", line 501, in wrap_socket

    return self.sslsocket_class._create(

  File "/Users/makotoohira/miniconda3/envs/acdc/lib/python3.9/ssl.py", line 1041, in _create

    self.do_handshake()

  File "/Users/makotoohira/miniconda3/envs/acdc/lib/python3.9/ssl.py", line 1310, in do_handshake

    self._sslobj.do_handshake()

ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1129)


During handling of the above exception, another exception occurred:


Traceback (most recent call last):

  File "/Users/makotoohira/miniconda3/envs/acdc/lib/python3.9/site-packages/requests/adapters.py", line 486, in send

    resp = conn.urlopen(

  File "/Users/makotoohira/miniconda3/envs/acdc/lib/python3.9/site-packages/urllib3/connectionpool.py", line 798, in urlopen

    retries = retries.increment(

  File "/Users/makotoohira/miniconda3/envs/acdc/lib/python3.9/site-packages/urllib3/util/retry.py", line 592, in increment

    raise MaxRetryError(_pool, url, error or ResponseError(cause))

urllib3.exceptions.MaxRetryError: HTTPSConnectionPool(host='hmgubox2.helmholtz-muenchen.de', port=443): Max retries exceeded with url: /index.php/s/CaMdYXiwxxoq3Ts/download/TimeLapse_2D.zip (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1129)')))


During handling of the above exception, another exception occurred:


Traceback (most recent call last):

  File "/Users/makotoohira/miniconda3/envs/acdc/lib/python3.9/site-packages/cellacdc/help/welcome.py", line 49, in run

    self.exp_path = myutils.download_examples(

  File "/Users/makotoohira/miniconda3/envs/acdc/lib/python3.9/site-packages/cellacdc/myutils.py", line 855, in download_examples

    download_url(

  File "/Users/makotoohira/miniconda3/envs/acdc/lib/python3.9/site-packages/cellacdc/myutils.py", line 1161, in download_url

    response = requests.get(url, stream=True, timeout=20)

  File "/Users/makotoohira/miniconda3/envs/acdc/lib/python3.9/site-packages/requests/api.py", line 73, in get

    return request("get", url, params=params, **kwargs)

  File "/Users/makotoohira/miniconda3/envs/acdc/lib/python3.9/site-packages/requests/api.py", line 59, in request

    return session.request(method=method, url=url, **kwargs)

  File "/Users/makotoohira/miniconda3/envs/acdc/lib/python3.9/site-packages/requests/sessions.py", line 589, in request

    resp = self.send(prep, **send_kwargs)

  File "/Users/makotoohira/miniconda3/envs/acdc/lib/python3.9/site-packages/requests/sessions.py", line 703, in send

    r = adapter.send(request, **kwargs)

  File "/Users/makotoohira/miniconda3/envs/acdc/lib/python3.9/site-packages/requests/adapters.py", line 517, in send

    raise SSLError(e, request=request)

requests.exceptions.SSLError: HTTPSConnectionPool(host='hmgubox2.helmholtz-muenchen.de', port=443): Max retries exceeded with url: /index.php/s/CaMdYXiwxxoq3Ts/download/TimeLapse_2D.zip (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:1129)')))

zsh: abort      acdc

(acdc) ***@***.*** ~ %



##############



Makoto Ohira
Rhind Lab
https://www.umassmed.edu/rhindlab/
Department of Biochemistry and Molecular Biotechnology
University of Massachusetts Chan Medical School
LRB 940D
364 Plantation Street
Worcester, MA 01655

________________________________

**Comment by ElpadoCan** [comment]
Oh jeez, that error happens because Cell-ACDC is trying to download the example data automatically, but it fails because the connection to our cloud times out. The reason is that the data is on our campus cloud (we are in Helmholtz Munich), but the Helmholtz centre was the victim of a massive hacker attack some months ago and many systems are still down.

I will think of a solution first thing tomorrow morning. In the meantime, you can download the timelapse example data from [here ](https://hmgubox2.helmholtz-muenchen.de/index.php/s/CaMdYXiwxxoq3Ts/download/TimeLapse_2D.zip) and the z-stacks from [here](https://hmgubox2.helmholtz-muenchen.de/index.php/s/CXZDoQMANNrKL7a/download/Yeast_Analysed_multi3D_zStacks.zip).

Note that if you need more test data we published the entire dataset that we used in the paper [here](https://zenodo.org/record/6795124).

Let me know if it works, thanks!

**Comment by makotojohira** [comment]
Fantastic - thank you so much!

Makoto Ohira
Rhind Lab
https://www.umassmed.edu/rhindlab/
Department of Biochemistry and Molecular Biotechnology
University of Massachusetts Chan Medical School
LRB 940D
364 Plantation Street
Worcester, MA 01655

________________________________

**Comment by ElpadoCan** [comment]
Hi Makoto,

I just released a new version with a fix for this. Note that this is a pre-release. To install it, activate the `acdc` environment and run the command `pip install cellacdc==1.4.0rc1`. If you are still having issues, try recreating the environment from scratch. To do so, deactivate the environment with `conda deactivate`, then remove the environment with `conda env remove -n acdc` and start from scratch.

Here are some more details of why you got those errors.

Unfortunately, the normal installation of `v1.3.0` of Cell-ACDC onto Mac M1 or M2 does not work. You can see another discussion [here](https://github.com/SchmollerLab/Cell_ACDC/discussions/320).

The reason is that the installation of the GUI library `PyQt5` from pip on Apple Silicon Macs does not work and I don't think the developer wants to fix that. There is a workaround by installing Cell-ACDC without dependencies and installing PyQt5 from conda with `conda install pyqt`, but I implemented a much better solution. 

Basically, since v1.4.0, Cell-ACDC uses `PyQt6` by default, while it will still support `PyQt5`. I have a Mac M1 myself and `PyQt6` installation from pip seems to work without issues.

Note that while Cell-ACDC is installed from pip, I still recommend using conda to create the environment. 

Long story short, I just released `v1.4.0rc1` which implements the fix I mentioned. Note that I would have liked to test this version a bit longer before releasing it but I don't think Apple Silicon users can wait any longer. This is also the reason why it's called `rc1` which means `release-candidate`. 

Whenever you have time, please, report any issue you find so that we get closer to a stable `1.4.0` release. 

Let me know how it goes and thank you very much for your patience!

Best,
Francesco

**Comment by makotojohira** [comment]
Thank you!   Will do!

Makoto Ohira
Rhind Lab
https://www.umassmed.edu/rhindlab/
Department of Biochemistry and Molecular Biotechnology
University of Massachusetts Chan Medical School
LRB 940D
364 Plantation Street
Worcester, MA 01655

________________________________
